src/cdk/layout/breakpoints-observer.ts

```diff
@@ -80,16 +80,12 @@ export class BreakpointObserver implements OnDestroy {
     stateObservable = concat(
       stateObservable.pipe(take(1)),
       stateObservable.pipe(skip(1), debounceTime(0)));
-    return stateObservable.pipe(map((breakpointStates: InternalBreakpointState[]) => {
-      const response: BreakpointState = {
-        matches: false,
-        breakpoints: {},
-      };
-      breakpointStates.forEach((state: InternalBreakpointState) => {
-        response.matches = response.matches || state.matches;
-        response.breakpoints[state.query] = state.matches;
-      });
-      return response;
+    return stateObservable.pipe(map(breakpointStates => {
+      return breakpointStates.reduce((previousState, {matches, query}) => ({
+        ...previousState,
+        breakpoints: {...previousState.breakpoints, [query]: matches},
```

Comment:
This is shorter, but less efficient since it transpiles to two `Object.assign` calls. I'm guessing that's why it was previously written with a `forEach`.

Reply:
Ok, reverted.